ui/src/shared/reducers/sources.js

```diff
@@ -1,4 +1,8 @@
-export default function sources(state = [], action) {
+const getInitialState = () => []
```

Comment:
That's a good point, for immutability's sake. We don't want to accidentally mutate the initialState. If it's always produced within a sort of initialState factory function, mutation will never happen. Good call!